### PR TITLE
adds rating functionality

### PR DESCRIPTION
**Pull request by skeltont**
### adds ability for logged in users to leave a rating.

an unrated route, both overall and by the user
screen shot 2016-03-24 at 4 19 17 pm

after rating
screen shot 2016-03-24 at 4 19 30 pm

### CODE REVIEW

File: app/assets/javascripts/rating.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
...lect.call(this,c(),u()))},this.clear=function(){var e=i("userOptions");o("ratingValue",i("originalRatingValue")),o("ratingText",i("originalRatingText")),g(c()),p(u()),h(),e.onClear.call(this,c(),u())},this.destroy=function(){var e=c(),n=u(),a=i("userOptions");x(t.$widget.find("a")),t.$widget.remove(),l(),r(),t.$elem.show(),a.onDestroy.call(this,e,n)}}return t.prototype.init=function(t,n){return this.$elem=e(n),this.options=e.extend({},e.fn.barrating.defaults,t),this.options},t}();e.fn.barrating=function(n,r){return this.each(function(){var a=new t;if(e(this).is("select")||e.error("Sorry, this plugin only works with select fields."),a.hasOwnProperty(n)){if(a.init(r,this),"show"===n)return a.show(r);if(a.$elem.data("barrating"))return a.$widget=e(this).next(".br-widget"),a[n](r)}else{if("object"==typeof n||!n)return r=n,a.init(r,this),a.show();e.error("Method "+n+" does not exist on jQuery.barrating")}})},e.fn.barrating.defaults={theme:"",initialRating:null,showValues:!1,showSelectedRating:!0,reverse:!1,readonly:!1,fastClicks:!0,hoverState:!0,silent:!1,wrapperClass:"br-wrapper",onSelect:function(e,t,n){},onClear:function(e,t){},onDestroy:function(e,t){}},e.fn.barrating.BarRating=t});
```
Comment (jonahgeorge) [inline app/assets/javascripts/rating.js]: Use `rails-assets` or a Ruby gem to import this

File: app/assets/javascripts/rating.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        theme: 'bootstrap-stars',
+        readonly: true,
+      });
+      if ($('#route_show_rating').data() !== undefined) {
+        rating = $('#route_show_rating').data().rating;
+      }
+      $('#route_show_rating').barrating('set',rating);
+   });
+
+  $(function() {
```
Comment (jonahgeorge) [inline app/assets/javascripts/rating.js]: Use the `$(function()` from above

File: app/controllers/ratings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class RatingsController < ApplicationController
+  def create
+    @rating = Rating.find_by(route_id: params[:route_id], user_id: current_user.id)
```
Comment (jonahgeorge) [inline app/controllers/ratings_controller.rb]: Use `Rating.find_or_create_by`

File: app/views/routes/show.html.erb [hunk 4]
```diff
@@ -44,6 +44,23 @@
...
             <td><%= @route.expiration_date.strftime("%b %d, %Y") %></td>
           </tr>
+          <tr>
+            <th><label for="rating">Overall Rating</label></th>
+            <td>
+              <% if @rating == 0 %>
+                <i>not yet rated</i>
+              <% else %>
+                <select id="route_show_rating" data-rating="<%= @rating %>">
+                  <option value="1">1</option>
```
Comment (jonahgeorge) [inline app/views/routes/show.html.erb:54]: ~~As an `enum` on the Rating model~~
Maybe not, need to look into this more

File: app/controllers/routes_controller.rb [hunk 5]
```diff
@@ -16,6 +16,18 @@ def show
     @request = MaintenanceRequest.new
     @comment = Comment.new
     @comments = @route.comments.most_recent
+
+    @user_rating = Rating.find_by(route_id: @route.id, user_id: current_user.id)
+    @rating = 0
+    ratings = Rating.where(route_id: @route.id).all
+    if ratings.size > 0
```
Comment (jonahgeorge) [inline app/controllers/routes_controller.rb]: Export this to a function on `Route`?

File: app/views/routes/show.html.erb (hunk 4 above)
Comment (skeltont) [inline app/views/routes/show.html.erb:54]: yeah it won't _really_ work, because of the nature of the plugin.

**Pull request merged**